scripts/androidBeforeInstall.js

```diff
@@ -1,43 +1,42 @@
 module.exports = function(ctx) {
-    var fs = ctx.requireCordovaModule('fs'),
-        path = ctx.requireCordovaModule('path'),
+    var fs = require('fs'),
+        path = require('path'),
         os = require("os"),
-        readline = require("readline"),
-        deferral = ctx.requireCordovaModule('q').defer();
+        readline = require("readline");
 
     var platformRoot = path.join(ctx.opts.projectRoot, 'www');
     var settingsFile = path.join(platformRoot, 'google-services.json');
 
     var platformAndroid = 'platforms/android';
 
-    if (fs.existsSync('platforms/android/app')) {
+    if (fs.existsSync('platforms/android/app/src/main')) {
```

Comment:
I'm confused about this change. If the /src/main exists the we should use `'platforms/android/app'` or it should be update also?

Reply:
I had some experiences where some plugins creates an "app" folder. In that case, I would not be able to identify whether cordova-android >= 7.x or not.
The condition is only to try to ensure this. Maybe have another way to do it.